visualization/app/codeCharta/ui/codeMap/codeMap.arrow.service.ts

```diff
@@ -8,12 +8,15 @@ import { SettingsService } from "../../state/settingsService/settings.service"
 import { ColorConverter } from "../../util/color/colorConverter"
 import { CodeMapBuilding } from "./rendering/codeMapBuilding"
 
-export class CodeMapArrowService implements BuildingHoveredSubscriber, BuildingUnhoveredSubscriber {
+export class CodeMapArrowService
+	implements BuildingSelectedEventSubscriber, BuildingDeselectedEventSubscriber, BuildingHoveredSubscriber, BuildingUnhoveredSubscriber {
 	private VERTICES_PER_LINE = 5
 	private map: Map<String, Node>
 	private arrows: Object3D[]
 	private isHovered: boolean = false
+	private isClicked: boolean = false
 	private hoveredNode: Node
+	private clickedNode: Node = null
```

Comment:
clickedNode is the same as selectedNode. This property is stored in the `threeSceneService` and we only want one single source of truth. Please use `getSelectedBuilding()` in ThreeSceneService instead of declaring another property in a service, that has nothing to do with the selection.

Reply:
or just pass on the building that you recieve from the onBuildingSelected event